pkg/sql/catalog/hydrateddesc/hydrate.go

```diff
@@ -82,6 +85,12 @@ func MakeTypeLookupFuncForHydration(
 		case catalog.TypeDescriptor:
 			typ = t
 		case catalog.TableDescriptor:
+			if !t.Hydrated() {
+				if err := Hydrate(ctx, t, typeLookupFunc); err != nil {
+					return tree.TypeName{}, nil, err
+				}
+				c.UpsertDescriptor(t)
```

Comment:
Should we also upsert `c` with all the other descriptors we look up, now that this function is recursive?

Reply:
good idea, I think we can do that.